kernel.go

```diff
@@ -106,6 +113,12 @@ func runKernel(connectionFile string) {
 		log.Fatal(err)
 	}
 
+	// channelsWG waits for all channel handlers to shutdown.
+	var channelsWG sync.WaitGroup
```

Comment:
Ok, I've thought a little about this WaitGroup and sequence of things around the heartbeat. It just doesn't feel quite like the right pattern to me. I think a couple modifications would improve this flow:

- The `quit` channel in `runHeartbeat` should not be a bool chan and/or we should use the context package to cancel.  See [here](https://www.goinggo.net/2017/10/the-behavior-of-channels.html) under `Signaling Without Data`: "If you choose to use your own channel for cancellation, rather than the context package, your channel should be of type chan struct{}. It is the zero-space, idiomatic way to indicate a channel used only for signalling."

- Why are we returning the close function from `runHeartBeat`? Seems like it would be more simple and readable if we just return the channel for cancellation.

- I don't think we actually need the `WaitGroup` do we? We will be blocked on the infinite message polling loop. Then if we receive a shutdown request, it will be handled with `handleShutdownRequest` ending in `os.Exit(0)`. Thus, I don't think we will ever get to this point will we? Do we need to instead pass the return cancellation channel (or context if we use that) to `handleShutdownRequest` somehow to make sure that this heartbeat goroutine exits gracefully?